### PR TITLE
Integrate opam-admin functionality into opam

### PR DESCRIPTION
**Pull request by AltGr**

### CODE REVIEW

File: src/client/opamAdminCommand.ml [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+let admin_command_doc =
+  "Tools for repository administrators"
+
+let admin_command_man = [
+  `S "DESCRIPTION";
+  `P "This command can perform various actions on repositories in the opam \
+      format. It is expected to be run from the root of a repository, i.e. a \
+      directory containing a 'repo' file and a subdirectory 'packages/' \
+      holding package definition within sub-subdirectories. A 'compilers/' \
```
Comment (avsm) [inline src/client/opamAdminCommand.ml]: s/sub-subdir/subdir

File: src/client/opamAdminCommand.ml [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    else match create_mirror with
+      | None ->
+        OpamAdminRepoUpgrade.do_upgrade ();
+        if OpamFilename.exists (OpamFilename.of_string "index.tar.gz") ||
+           OpamFilename.exists (OpamFilename.of_string "urls.txt")
+        then
+          OpamConsole.note
+            "Indexes need updating: you should now run:\n\
+             \n\
+            \  opam admin index"
```
Comment (avsm) [inline src/client/opamAdminCommand.ml]: is this backslash intended to be here?

Comment (AltGr) [inline src/client/opamAdminCommand.ml]: Well, yes, for indentation ?

Comment (avsm) [inline src/client/opamAdminCommand.ml]: oops, thought it was a bad escape in my diff

**Comment by AltGr** [comment]
Merging this will _remove_ the opam-admin tool from the opam source and targets.
It may break some scripts... @avsm, would that be ok for you ?

**Comment by avsm** [comment]
lgtm!

**Pull request merged**